internal/api/impl/auth/auth.go

```diff
@@ -58,6 +60,23 @@ func getRedirectURI(r *http.Request, authKind string, slugID string) string {
 	return rd.String()
 }
 
+// newHTTPClient is a simple http client builder designed to be used for the queries to external authentication providers.
+func newHTTPClient(httpConfig config.HTTP) (*http.Client, error) {
+	httpClient := &http.Client{
+		Timeout: time.Duration(httpConfig.Timeout),
+	}
+	if httpConfig.TLSConfig != nil {
+		tlsConfig, err := promConfig.NewTLSConfig(httpConfig.TLSConfig)
+		if err != nil {
+			return nil, err
+		}
+		if tlsConfig != nil {
+			httpClient.Transport = &http.Transport{TLSClientConfig: tlsConfig}
```

Comment:
I think something like that will be cleaner:

https://github.com/perses/perses/blob/release/v0.48/pkg/client/config/config.go#L157-L165

Reply:
would be great to be able to use TLS configuration for oauth authentication in the client as well. If you can find the way, that would be cool !

Also I think it will be nice to have it in v0.48.0

Reply:
The client doesn't make direct queries to the oauth/oidc provider. These queries are done by the backend exactly in the code I modified. 

Or are you taking about a TLS config to query the perses backend? I think this is already configurable no?

Reply:
ah interesting, I forgot Perses was doing the request to the provider.

But yeah also the TLS config for querying the perses backend. And actually when we are using oauth, I didn't find the way to inject the tls config into the http client of the oauth client.

So if you know how to do it, that would be great !

Reply:
This is done in the newQueryContext function of that PR. I attached in the comment a GitHub issue for reference.

Reply:
Here is the ref 
https://github.com/golang/oauth2/issues/187

Reply:
ah yeah I see. Thanks, I will send a PR to modify the perses client